src/snapred/backend/data/LocalDataService.py

```diff
@@ -303,230 +253,85 @@ def _constructReductionDataRoot(self, runNumber: str, useLiteMode: bool) -> Path
-        else:
-            return max(zip(things, otherThings), default=(self.VERSION_START, None))
+    def readCalibrationIndex(self, runId: str, useLiteMode: bool):
+        return self.calibrationIndexor(runId, useLiteMode).getIndex()
 
-    def _getFileOfVersion(self, fileRegex: str, version: int):
-        foundFiles = self._findMatchingFileList(fileRegex, throws=False)
-        fileVersions = [self._extractFileVersion(file) for file in foundFiles]
-        where = fileVersions.index(version)
-        return foundFiles[where]
+    def readNormalizationIndex(self, runId: str, useLiteMode: bool):
+        return self.normalizationIndexor(runId, useLiteMode).getIndex()
 
-    def _getLatestFile(self, fileRegex: str):
-        foundFiles = self._findMatchingFileList(fileRegex, throws=False)
-        fileVersions = [self._extractFileVersion(file) for file in foundFiles]
-        _, latestFile = self._getLatestThing(fileVersions, otherThings=foundFiles)
-        return latestFile
+    def readReductionIndex(self, runId: str, useLiteMode: bool):
+        # NOTE currently nothing in the code needs this, but it may as well exist
+        return self.reductionIndexor(runId, useLiteMode).getIndex()
 
-    def _getLatestVersionNumber(self, versionRoot: Path) -> int:
-        """
-        Examine the filesystem to get the latest version number.
-        """
-        versionPathGlob = str(versionRoot / "v_*")
-        versionDirs = self._findMatchingDirList(versionPathGlob, throws=False)
-        versions = [self._extractFileVersion(dir_) for dir_ in versionDirs]
-        return self._getLatestThing(versions)
-
-    def _getLatestCalibrationVersionNumber(self, stateId: str, useLiteMode: bool) -> int:
-        """
-        Ignoring the calibration index, get the version number of the latest set of calibration files.
-        """
-        return self._getLatestVersionNumber(self._constructCalibrationStatePath(stateId, useLiteMode))
+    def writeCalibrationIndexEntry(self, entry: CalibrationIndexEntry, version: Optional[int] = None):
+        self.calibrationIndexor(entry.runNumber, entry.useLiteMode).addIndexEntry(entry, version)
 
-    def _getLatestNormalizationVersionNumber(self, stateId: str, useLiteMode: bool) -> int:
-        """
-        Ignoring the normalization index, get the version number of the latest set of normalization files.
-        """
-        return self._getLatestVersionNumber(self._constructNormalizationStatePath(stateId, useLiteMode))
+    def writeNormalizationIndexEntry(self, entry: NormalizationIndexEntry, version: Optional[int] = None):
+        self.normalizationIndexor(entry.runNumber, entry.useLiteMode).addIndexEntry(entry, version)
 
-    def _getLatestReductionVersionNumber(self, runNumber: str, useLiteMode: bool) -> int:
-        """
-        Get the version number of the latest set of reduction data files.
-        """
-        return self._getLatestVersionNumber(self._constructReductionDataRoot(runNumber, useLiteMode))
```

Comment:
These methods handled extracting version numbers from the directory tree, and are no longer needed.  Similar logic is replicated inside `Indexor`